File: vigenere_cipher.py
from caesar_cipher import switch_alphabet_root

ALPHABET = "abcdefghijklmnopqrstuvwxyz"
COLORS = {
    'RED': '\033[31m',
    'GREEN': '\033[32m',
    'YELLOW': '\033[33m',
    'MAGENTA': '\033[35m',
    'RESET': '\033[0;0m'
}

def generate_table():
    table = list()
    for times in range(0, 26):
        table.append(switch_alphabet_root(times))
    return table


def encrypt(key, message):
    encrypted_message = list()

    for index in range(0, len(key)):
        x = ALPHABET.index(key[index])
        y = ALPHABET.index(message[index])
        encrypted_message.append(switch_alphabet_root(y)[x])

    return "".join(encrypted_message)


print(encrypt('bom', 'bom'))

File: src/scripts/vigenereCipher.js
const alphabet = require("../utils/alphabet");
const switchAlphabetRoot = require("../utils/switchAlphabetRoot");

function generateTable() {
    table = [];
    for (let index = 0; index < 26; index++) {
        table.push(switchAlphabetRoot(index));
    }

    return table;
};

function encrypt(key, message) {
    const encryptedMessage = [];

    for (let index = 0; index < key.length; index++) {
        let x = alphabet.indexOf(key[index]);
        let y = alphabet.indexOf(message[index]);
        encryptedMessage.push(switchAlphabetRoot(y)[x]);
    };

    return encryptedMessage.join("");
};

function decrypt(key, message) {
    const table = generateTable();
    const encryptedMessage = [];
    let x, y = 0;

    for (let i = 0; i < key.length; i++) {
        x = alphabet.indexOf(key.charAt(i));
        
        for (let j = 0; j < alphabet.length; j++) {
            if (table[j].charAt(x) == message.charAt(i)) {
                y = j;
            };
        };

        encryptedMessage[i] = switchAlphabetRoot(y).charAt(0);
    };

    return encryptedMessage.join("");
};
